### ISSUE TITLE
usernames cannot contain umlauts

### ISSUE THREAD

**Issue by diosmosis** [piwik/piwik#7114]
Caused by line https://github.com/piwik/piwik/blob/master/core/Piwik.php#L601. The regex prevents usernames from having special characters. I'm not sure what exactly the point of this is, why not allow all characters in a login handle?

Anyway, this is an issue for at least one user using LoginLdap: https://github.com/piwik/plugin-LoginLdap/issues/86

**Comment by tsteur** [comment]
FYI: If we change it, we need to check all usages of the login for security etc. It might be used somewhere and not proper escaped as one was assuming it contains only `[a-Z0-9-_]`

**Comment by ThaDafinser** [comment]
@tsteur is it possible to check if it's a character or a special char?
Didnt found something until yet...i use currently `\p` should contain all characters but does not...
```
[\p{L}0-9äöüÄÖÜßé]
```

**Comment by tsteur** [comment]
If this is just about umlauts as a temporary fix we should be able to just add `äöüÄÖÜßé` to the allowed characters. It will of course not solve it entirely but would maybe make things easier for some users :) I read about `\p{L} ` but not sure which letters it actually contains and whether there are possible security issues etc.

**Comment by ThaDafinser** [comment]
@tsteur depends on the charset...used it somewhere for LDAP regex, but also didnt contain all wanted chars :-)

Seems to be a bit complicated....

http://www.regular-expressions.info/unicode.html#script

**Comment by mattab** [comment]
if we can fix this in core by adding umlauts to the regex then maybe we should just do this for now?

**Comment by tsteur** [comment]
We should just still leave the issue open or create a follow up issue to also fix this for other letters. People will most likely have this problem again very soon

**Comment by ChristianGfK** [comment]
Is there any workaround for this?
In combination with LoginLdap and a company-wide Active Directory, this is kind of bad. :|

**Comment by ThaDafinser** [comment]
@ChristianGfK try this workaround (NOT tested)

Replace this line https://github.com/piwik/piwik/blob/master/core/Piwik.php#L592

```php
&& (preg_match('/^[A-Za-zÄäÖöÜüß0-9_.@+-]*$/D', $userLogin) > 0))
```

**Comment by ChristianGfK** [comment]
Thanks @ThaDafinser, I'll have a look at doing that when I take the 2.16 plunge later. :-)

**Comment by ChristianGfK** [comment]
test

**Comment by mattab** [comment]
as we have so many German users, let's do the quick fix of allowing german characters in 2.16.x :+1:
